core/app/models/spree/taxon.rb

```diff
@@ -62,13 +62,15 @@ class Taxon < Spree::Base
 
     self::Translation.class_eval do
       alias_attribute :slug, :permalink
-
-      before_create :set_permalink
+      before_update :set_permalink
 
       def set_permalink
         parent = translated_model.parent
         name_with_fallback = name || translated_model.name
 
+        if self.name.blank? & self.permalink.blank?
+          self.permalink = translated_model.name.to_url
+        end
         if parent.present?
           self.permalink = [parent.permalink, (permalink.blank? ? name_with_fallback.to_url : permalink.split('/').last)].join('/')
         else
```

Comment:
You first set permalink to:
```rb
translated_model.name.to_url
```
then to
```rb
[parent.permalink, (permalink.blank? ? name_with_fallback.to_url : permalink.split('/').last)].join('/')
```

The second will overwrite the other. Is this intentional? Is this missing and `elsif`